embassy-stm32/src/cordic/mod.rs

```diff
@@ -0,0 +1,804 @@
+//! coordinate rotation digital computer (CORDIC)
+
+use embassy_hal_internal::drop::OnDrop;
+use embassy_hal_internal::{into_ref, Peripheral, PeripheralRef};
+
+use crate::{dma, peripherals};
+
+mod enums;
+pub use enums::*;
+
+mod errors;
+pub use errors::*;
+
+pub mod utils;
+
+pub(crate) mod sealed;
+
+/// Low-level CORDIC access.
+#[cfg(feature = "unstable-pac")]
+pub mod low_level {
+    pub use super::sealed::*;
+}
+
+const INPUT_BUF_MAX_LEN: usize = 16;
```

Comment:
When using DMA mode, the driver need a buf to merge ARG1 list and ARG2 list into one DMA list.

For current implementation, Cordic driver allocates a fixed size of memeory for it, should we instead ask user to provide a buffer size for Cordic driver? If so, how should we design the API?

Reply:
i'd ask the user to pass the data already interleaved, in `[a1, b1, a2, b2, ...]` format. This way avoids copies if they can arrange for the data to be written that way when it's created. If they can't then they have to interleave it themselves, which allows them to choose how the buffer is allocated and sized.